Sources/Kitura/bodyParser/BodyParser.swift

```diff
@@ -236,22 +239,22 @@ public class BodyParser: RouterMiddleware {
                         //TODO: Deal with this
                         currentPart.headers[.transferEncoding] = bodyLineAsString
                     }
-                    else if !bodyLineAsString.isEmpty {
-                        // is data, add to data object
-                        if partData.length > 0 {
-                            // data is multiline, add linebreaks back in
-                            partData.append(newLineData)
-                        }
-                        partData.append(bodyLine)
+                    else if bodyLineAsString.isEmpty {
+                        // empty line signals the end of headers
+                        state = .body
                     }
+                    // skip custom headers
```

Comment:
Custom headers should still be retained and passed along, we should either expand the Header type to include .other(String) where the associated value is the unsupported header.